drivers/flash/flash_rpc/flash_rpc_controller.c

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/drivers/flash.h>
+#include <zephyr/logging/log.h>
+#include <drivers/flash/flash_rpc.h>
+
+#include <nrf_rpc/nrf_rpc_ipc.h>
+#include <nrf_rpc_cbor.h>
+
+#include <zcbor_common.h>
+#include <zcbor_decode.h>
+#include <zcbor_encode.h>
+
+#ifndef CONFIG_FLASH_RPC_SYS_INIT
+LOG_MODULE_REGISTER(FLASH_RPC, CONFIG_FLASH_RPC_LOG_LEVEL);
+#else
+LOG_MODULE_DECLARE(FLASH_RPC, CONFIG_FLASH_RPC_LOG_LEVEL);
+#endif
+
+#define CBOR_BUF_FLASH_MSG_SIZE (sizeof(void *) + sizeof(size_t) + sizeof(off_t) + 32)
+
+NRF_RPC_IPC_TRANSPORT(flash_rpc_api_tr, DEVICE_DT_GET(DT_NODELABEL(ipc0)), "flash_rpc_api_ept");
+NRF_RPC_GROUP_DEFINE(flash_rpc_api, "flash_rpc_api", &flash_rpc_api_tr, NULL, NULL, NULL);
+
+#if DT_NODE_HAS_STATUS(DT_INST(0, nordic_rpc_flash_controller), okay)
+#define DT_DRV_COMPAT nordic_rpc_flash_controller
+#else
+#error "Node is not available"
+#endif
+
+#define FLASH_RPC_CONTROLLER_NODE DT_INST(0, nordic_rpc_flash_controller)
+#define FLASH_RPC_NODE DT_INST_CHILD(0, flash_rpc_0)
+#define FLASH_RPC_FLASH_SIZE DT_REG_SIZE(FLASH_RPC_NODE)
+#define FLASH_RPC_ERASE_UNIT DT_PROP(FLASH_RPC_NODE, erase_block_size)
+#define FLASH_RPC_PROG_UNIT DT_PROP(FLASH_RPC_NODE, write_block_size)
+#define FLASH_RPC_FLASH_SIZE DT_REG_SIZE(FLASH_RPC_NODE)
+
+#define FLASH_RPC_PAGE_COUNT (FLASH_RPC_FLASH_SIZE/FLASH_RPC_ERASE_UNIT)
+
+static const struct flash_parameters flash_rpc_parameters = {
+	.write_block_size = FLASH_RPC_PROG_UNIT,
+	.erase_value = 0xff,
+};
+
+void rsp_error_code_handle(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+			   void *handler_data)
+{
+	int32_t val;
+
+	if (zcbor_int32_decode(ctx->zs, &val)) {
+		*(int *)handler_data = (int)val;
+	} else {
+		*(int *)handler_data = -NRF_EINVAL;
+	}
+}
+
+static void flash_rpc_get_rsp(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+				 void *handler_data)
+{
+	int *result = (int *)handler_data;
+
+	if (!zcbor_int32_decode(ctx->zs, result)) {
+		LOG_ERR("Unable to decode result");
+		*result = -EINVAL;
+	}
+}
+
+static bool encode_flash_msg(struct nrf_rpc_cbor_ctx *ctx, off_t *offset, void *ptr, size_t *len)
+{
+	NRF_RPC_CBOR_ALLOC(&flash_rpc_api, *ctx, CBOR_BUF_FLASH_MSG_SIZE);
+
+	if (!zcbor_uint32_put(ctx->zs, (uint32_t)*offset)) {
+		return false;
+	}
+
+	if (!zcbor_uint32_put(ctx->zs, (uint32_t)ptr)) {
+		return false;
+	}
+
+	return zcbor_uint32_put(ctx->zs, (uint32_t)*len);
```

Comment:
You can probably just use the && between calls, the way we do in MCUmgr while decoding zcbor.

Reply:
ok, I just don't prefer that style and I assumed the compiler would optimize it away anyways :) but I've changed it to your suggestion.